src/workers/LoadWorkerHandle.ts

```diff
@@ -0,0 +1,207 @@
+import { ImageInfo } from "../Volume";
+import { CreateLoaderOptions, VolumeFileFormat, pathToFileType } from "../loaders";
+import { ThreadableVolumeLoader, LoadSpec, RawChannelDataCallback, VolumeDims } from "../loaders/IVolumeLoader";
+import { TiffLoader } from "../loaders/TiffLoader";
+import {
+  WorkerMsgType,
+  WorkerRequest,
+  WorkerRequestPayload,
+  WorkerResponse,
+  WorkerResponsePayload,
+  ChannelLoadEvent,
+  WorkerResponseKind,
+} from "./types";
+import { rebuildImageInfo, rebuildLoadSpec } from "./util";
+
+type StoredPromise<T extends WorkerMsgType> = {
+  type: T;
+  resolve: (value: WorkerResponsePayload<T>) => void;
+  reject: (reason?: unknown) => void;
+};
+
+/**
+ * A handle that holds the worker and lets us interact with it through async calls and events rather than messages.
+ * This is separate from `LoadWorker` so that `sendMessage` and `onChannelData` can be shared with `WorkerLoader`s
+ * without leaking the API outside this file.
+ */
+class SharedLoadWorkerHandle {
+  private worker: Worker;
+  private pendingRequests: (StoredPromise<WorkerMsgType> | undefined)[] = [];
+  private workerOpen = true;
+
+  public onChannelData: ((e: ChannelLoadEvent) => void) | undefined = undefined;
```

Comment:
where is this used? Not sure why it is stored here.

Reply:
The idea of this class is to track outgoing messages to the worker, and to convert the stream of incoming messages from the worker into resolved `Promise`s and triggered events. This is meant to work a bit like attaching an event listener: `WorkerLoader`s can set this property to handle data loads (recall that only one `WorkerLoader` can be "active" at a time, so we shouldn't get collisions, though we do still check that messages received are meant for the current loader). It's called by this class anytime it receives a message from the worker with `responseKind: WorkerResponseResult.EVENT`.